src/components/Task.js

```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faTrash,
+  faEdit,
+  faArrowRight,
+  faArrowLeft,
+} from "@fortawesome/free-solid-svg-icons";
+
+import TaskUpdate from "./TaskUpdate";
+
+class Task extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isEditing: false,
+      title: this.props.title, // Initialize title in the local state
+      task: this.props.task, // Initialize task in the local state
+      timeTaken: this.props.timeTaken,
+      keyTakeaways: this.props.keyTakeaways,
+    };
+  }
+
+  handleUpdateClick = () => {
+    this.setState({ isEditing: true });
+  };
+
+  handleUpdateComplete = () => {
+    this.setState({ isEditing: false });
+  };
+
+  // ... (delete logic, if needed)
+  handleDeleteClick = () => {
+    this.props.deleteTask(this.props.id); // Calls the deleteTask function defined in TaskList.js
+  };
+
+  handleMoveTaskOpenClick = () => {
+    this.props.moveTaskOpen(this.props.id);
+  };
+
+  handleMoveTaskInProgressClick = () => {
+    this.props.moveTaskInProgress(this.props.id);
+  };
+
+  handleMoveTaskInProgressBackToOpen = () => {
+    this.props.moveTaskInProgressBacktoOpen(this.props.id);
+  };
+
+  handleMoveTaskInReviewBackToInProgress = () => {
+    this.props.moveTaskInReviewBackToInProgress(this.props.id);
+  };
+
+  handleMoveTaskInReviewClick = () => {
+    this.props.moveTaskInReview(this.props.id);
+  };
+
+  handleReflectedTaskClick = () => {
+    this.props.reflectionToDo(this.props.id);
+  };
+
+  render() {
+    // showButton prop is received as a parameter in the 'render' method and is then destructured from this.props
+
+    const { showButton } = this.props;
+    const { showButtonInProgress } = this.props;
+    const { showButtonCompleted } = this.props;
+    const { showButtonInReview } = this.props;
```

Comment:
```suggestion
    const { showButton, showButtonInProgress, showButtonCompleted, showButtonInReview } = this.props;
```

Let's use destructuring like this, so we do not need to define a variable in a single line like we had to do before. Otherwise there is no benefit of using destructuring.